biojava-structure-gui/src/main/java/org/biojava/nbio/structure/gui/util/PDBUploadPanel.java

```diff
@@ -150,7 +151,8 @@ private Structure getStructure(JTextField filePath,JTextField chainId) throws St
 			throw new StructureException(e);
 		}
 
-		Structure reduced = StructureTools.getReducedStructure(s, chainId.getText());
+//		Structure reduced = StructureTools.getReducedStructure(s, chainId.getText());
+		Structure reduced = new SubstructureIdentifier(s.getPDBId().getId()+"."+ chainId.getText()).reduce(s);  //TODO double check this
```

Comment:
To avoid string parsing you can use
```
new SubstructureIdentifier(s.getPDBID(), Lists.asList(new ResidueRange(chainId.getText(), null, null))).reduce()
```

Reply:
But the concept of replacing `getReducedStructure` with `StructureIdentifier.reduce()` is correct

Reply:
Fine. Thank you

Reply:
OK Done